boards/arm/particle_boron/board.c

```diff
@@ -53,6 +53,7 @@ static int board_particle_boron_init(void)
 }
 
 /* needs to be done after GPIO driver init, which is at
- * POST_KERNEL:KERNEL_INIT_PRIORITY_DEFAULT.
+ * PRE_KERNEL_1:CONFIG_GPIO_INIT_PRIORITY, but before the
+ * modem driver (gsm_ppp or ublox-sara-r4).
  */
-SYS_INIT(board_particle_boron_init, POST_KERNEL, 99);
+SYS_INIT(board_particle_boron_init, POST_KERNEL, 41);
```

Comment:
let's not hardcode priorities like this, can you create a board level Kconfig option instead?

Reply:
Btw, can't this be solved using `regulator-fixed`?

Reply:
> Btw, can't this be solved using `regulator-fixed`?

Sounds like a good idea, I will try it out.

Reply:
I ended up removing more lines than I added, so that's good.

However the magic number 41 showed up again in `CONFIG_REGULATOR_FIXED_INIT_PRIORITY=41`. There are two other boards where this is done similarly. The `thingy53_nrf5340` board uses `BUILD_ASSERT` to check that nobody breaks the init order, let me know if you want something similar here.